docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_in_vpc_remediation.mdx

```diff
@@ -1,6 +1,205 @@
+   ```
+
+3. **Create a Cluster Subnet Group:**
+   Create a cluster subnet group that includes the subnets you created. This subnet group will be used when launching the Redshift cluster.
+   ```sh
+   aws redshift create-cluster-subnet-group --cluster-subnet-group-name my-subnet-group --description "My Redshift Subnet Group" --subnet-ids <subnet-id-1> <subnet-id-2>
+   ```
+
+4. **Launch the Redshift Cluster within the VPC:**
+   Finally, launch the Redshift cluster specifying the VPC subnet group.
+   ```sh
+   aws redshift create-cluster --cluster-identifier my-redshift-cluster --node-type dc2.large --master-username myuser --master-user-password mypassword --cluster-subnet-group-name my-subnet-group --number-of-nodes 2
+   ```
+
+By following these steps, you ensure that your Redshift clusters are launched within a VPC, thereby enhancing security and network control.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift clusters from being launched outside of a VPC using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that Redshift clusters are launched within a VPC:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a VPC**:
+   If you don't already have a VPC, you need to create one. This step ensures that you have a VPC to launch your Redshift cluster into.
+   ```python
+   import boto3
+
+   ec2 = boto3.client('ec2')
+
+   # Create a VPC
+   response = ec2.create_vpc(
+       CidrBlock='10.0.0.0/16'
+   )
+   vpc_id = response['Vpc']['VpcId']
+   print(f'Created VPC with ID: {vpc_id}')
+   ```
+
+3. **Create Subnets and Security Groups**:
+   Create subnets and security groups within the VPC to ensure that your Redshift cluster has the necessary network infrastructure.
+   ```python
+   # Create a subnet
+   subnet_response = ec2.create_subnet(
+       VpcId=vpc_id,
+       CidrBlock='10.0.1.0/24'
+   )
+   subnet_id = subnet_response['Subnet']['SubnetId']
+   print(f'Created Subnet with ID: {subnet_id}')
+
+   # Create a security group
+   sg_response = ec2.create_security_group(
+       GroupName='RedshiftSG',
+       Description='Security group for Redshift',
+       VpcId=vpc_id
+   )
+   sg_id = sg_response['GroupId']
+   print(f'Created Security Group with ID: {sg_id}')
+   ```
+
+4. **Launch Redshift Cluster within the VPC**:
+   Use the VPC ID, subnet ID, and security group ID to launch the Redshift cluster within the VPC.
+   ```python
+   redshift = boto3.client('redshift')
+
+   # Launch Redshift cluster within the VPC
+   cluster_response = redshift.create_cluster(
+       ClusterIdentifier='my-redshift-cluster',
+       NodeType='dc2.large',
+       MasterUsername='masteruser',
+       MasterUserPassword='MasterUserPassword1!',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>